### PR TITLE
Replace all Set usages of Project by List #7972

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectPermissionsJson.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ModifyProjectParams.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5268040&bid=17257511)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectPermissionsJson.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ModifyProjectParams.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java  -2
+ modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5268040&bid=17257511)

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java [hunk 1]
```diff
@@ -38,56 +36,39 @@ public ProjectJson( final Project project, final ProjectPermissions projectPermi
             projectPermissions.getViewer().getSet() ) ) : null;
     }
 
+    @SuppressWarnings("UnusedDeclaration")
```
Comment (rymsha) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java]: Don't we have tests for Json classes?

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectsJson.java [hunk 2]
```diff
@@ -13,6 +13,7 @@ public ProjectsJson( final List<ProjectJson> projects )
         this.projects = ImmutableList.copyOf( projects );
     }
 
+    @SuppressWarnings("UnusedDeclaration")
     public ImmutableList<ProjectJson> getProjects()
```
Comment (rymsha) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectsJson.java]: Should be List

File: modules/itest/itest-core/src/test/java/com/enonic/xp/core/project/ProjectServiceImplTest.java [hunk 3]
```diff
@@ -697,4 +697,13 @@ private Project doCreateProject( final ProjectName name, final ProjectPermission
         return project;
     }
 
+    private void checkProjectsEquality( final Project p1, final Project p2 )
+    {
+        assertEquals( p1.getName(), p2.getName() );
```
Comment (rymsha) [inline modules/itest/itest-core/src/test/java/com/enonic/xp/core/project/ProjectServiceImplTest.java]: use assertAll

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java (hunk 1 above)
Comment (vbradnitski) [reply modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java]: yes, added it dirrect checks few minutes ago, forgot to remove :)

Comment (vbradnitski) [reply modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java]: we have, but without using dirrect method calls.

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ModifyProjectParams.java  -3
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectPermissionsJson.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5268040&bid=17257511)

File: modules/itest/itest-core/src/test/java/com/enonic/xp/core/project/ProjectServiceImplTest.java [hunk 4]
```diff
@@ -697,4 +698,10 @@ private Project doCreateProject( final ProjectName name, final ProjectPermission
         return project;
     }
 
+    private void checkProjectsEquality( final Project p1, final Project p2 )
```
Comment (rymsha) [inline modules/itest/itest-core/src/test/java/com/enonic/xp/core/project/ProjectServiceImplTest.java]: IMO assertProjectEquals is good name

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectsJson.java [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 import com.google.common.collect.ImmutableList;
 
+@SuppressWarnings("UnusedDeclaration")
```
Comment (rymsha) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectsJson.java]: Should be covered with tests and then needs no SuppressWarnings

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java [hunk 6]
```diff
@@ -10,6 +8,7 @@
 import com.enonic.xp.project.Project;
 import com.enonic.xp.project.ProjectPermissions;
 
+@SuppressWarnings("UnusedDeclaration")
```
Comment (rymsha) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java]: Should be covered with tests and then needs no SuppressWarnings

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java  -2
+ modules/core/core-api/src/main/java/com/enonic/xp/project/ModifyProjectParams.java  -3
+ modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java  -3
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectPermissionsJson.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5268040&bid=17257511)

**Pull request merged**